python/src/aac/plugins/validators/unused_definitions/_validate_used_definitions.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from aac.lang.definitions.definition import Definition
+from aac.lang.definitions.references import get_definition_type_references_from_list
+from aac.lang.language_context import LanguageContext
+from aac.plugins.validators import ValidatorFindings, ValidatorResult
+
+
+PLUGIN_NAME = "Validate definition references"
+VALIDATION_NAME = "Definition references exist"
+
+
+def validate_used_definitions(
+    definition_under_test: Definition,
+    target_schema_definition: Definition,
+    language_context: LanguageContext,
+    *validation_args,
+) -> ValidatorResult:
+    """
+    Validates that the definition has valid definition references.
+
+    Args:
+        definition_under_test (Definition): The definition that's being validated.
+        target_schema_definition (Definition): A definition with applicable validation.
+        language_context (LanguageContext): The language context.
+
+    Returns:
+        A ValidatorResult containing any applicable error messages.
+    """
+    findings = ValidatorFindings()
+
+    if language_context.get_definition_by_name(target_schema_definition.name):
+        logging.debug(f"Valid definition: {target_schema_definition.name}.")
+
+        definitions_to_test = get_definition_type_references_from_list(target_schema_definition, definition_under_test)
+        if len(definitions_to_test) > 0:
+            logging.info(f"The definition {definition_under_test.name} references a valid definition {target_schema_definition.name}")
+        else:
+            logging.info(f"The definition {definition_under_test.name} does not reference another definition.")
+    else:
+        undefined_definition_error_message = f"Nonexistant definition: '{target_schema_definition.name}'."
+        reference_lexeme = target_schema_definition.get_lexeme_with_value(target_schema_definition.name)
+        logging.info(undefined_definition_error_message)
+
+        if reference_lexeme:
+            findings.add_error_finding(
+                definition_under_test, undefined_definition_error_message, PLUGIN_NAME, reference_lexeme
+            )
```

Comment:
Here is an adjusted solution for the validation. It took me a little to get back into how the validator works, so we can certainly do a better job with the description of these methods; with that in mind I expanded on the args docstring to give a better idea of the `target_schema_definition` arg.
```suggestion
    Args:
        definition_under_test (Definition): The definition that's being validated.
        target_schema_definition (Definition): The schema definition with the validations rules that trigger the validation.
        language_context (LanguageContext): The language context.

    Returns:
        A ValidatorResult containing any applicable error messages.
    """
    findings = ValidatorFindings()

    # Get references to the definition being validated
    referencing_definitions = get_definition_type_references_from_list(definition_under_test, language_context.definitions)

    if len(referencing_definitions) == 0:
        no_references_info_message = f"'{definition_under_test.name}' is not referenced by other definitions in the context."
        reference_lexeme = definition_under_test.get_lexeme_with_value(definition_under_test.name)
        logging.info(no_references_info_message)

        if reference_lexeme:
            findings.add_info_finding(
                definition_under_test, no_references_info_message, PLUGIN_NAME, reference_lexeme
            )

    return ValidatorResult([definition_under_test], findings)
```

With this validation we're checking for any references to `definition_under_test` in the language context -- which is accomplished with the line `referencing_definitions = get_definition_type_references_from_list(definition_under_test, language_context.definitions)`. After collecting of the other definitions referencing our definition being validated, we can check for a non-zero number of references. If there are no references, then we'll want to return our info finding result otherwise there is nothing to report to the user.

Reply:
Incorporated your suggestion, but still included the check for is the definition even in the context to begin with. I would think that would be something we'd want to know, trying to reference a definition that doesn't exist. 

Unless that is happening somewhere else already and I've missed it? I didn't see anything in the get references method that indicated it would be checked and I didn't spot any validators that were. The ones I thought might (defined references, reference fields, and reference targets) didn't seem to. Did I miss checking one?